homeassistant/components/zeroconf/__init__.py

```diff
@@ -557,7 +557,9 @@ def info_from_service(service: AsyncServiceInfo) -> ZeroconfServiceInfo | None:
     )
 
 
-def _first_non_link_local_or_v6_address(addresses: list[bytes]) -> str | None:
+def _first_non_link_local_or_v6_address(
```

Comment:
```suggestion
def _first_non_link_local_prefer_ipv4(
```

Reply:
I just removed the `or_v6` part and updated the comments. Let me know if that works for you.